src/java.base/share/classes/java/lang/Process.java

```diff
@@ -149,12 +169,114 @@ public abstract class Process {
     private Charset inputCharset;
     private BufferedReader errorReader;
     private Charset errorCharset;
+    private boolean closed;     // true if close() has been called
 
     /**
      * Default constructor for Process.
      */
     public Process() {}
 
+    /**
+     * Close all reader and writer streams and wait for the process to terminate.
+     * This method is idempotent, if this {@code Process} has already been closed
+     * invoking this method has no effect.
+     * <p>
+     * If the data from the process input or error streams is needed, it must be read before
+     * calling this method. The contents of streams that have not been read to end of stream
+     * are lost, they are discarded or ignored.
+     * <p>
+     * If the process exit value of the process is of interest, then the caller must
+     * {@linkplain #waitFor() wait for} the process to terminate before calling this method.
+     * <p>
+     * Streams should be closed when no longer needed.
+     * Closing an already closed stream usually has no effect but is specific to the stream.
+     * If an {@code IOException} occurs when closing a stream it is thrown
+     * after the process has terminated. {@linkplain Exception}s
+     * thrown by closing the streams, if any, are added to the first
+     * {@code IOException} as {@linkplain IOException#addSuppressed suppressed exceptions}.
+     * <p>
+     * After the streams are closed this method {@linkplain #waitFor() waits for} the
+     * process to terminate. If interrupted while {@linkplain #waitFor waiting} for termination
+     * the process is {@linkplain #destroyForcibly() forcibly destroyed} and
+     * this method continues to wait for the process to terminate.
+     * The interrupt status is re-asserted before this method returns and
+     * any {@code IOExceptions} are thrown.
+     * <p>
+     * Try-with-resources example to write text to a process, read back the
+     * response, and close the streams and process:
+     * {@snippet class=ProcessExamples region=example}
+     *
+     * @implSpec
+     * The {@code outputWriter} and {@code outputStream} to the process are closed.
+     * The {@code inputReader} and {@code inputStream} from the process are closed.
+     * The {@code errorReader} and {@code errorStream} from the process are closed.
```

Comment:
The implSpec is quite verbose as it lists all 6 streams and character streams. Maybe shorten to say that the method first closes all I/O streams, then waits for the process to terminate?